libipl/p10/ipl0.C

```diff
@@ -379,22 +392,163 @@ static int ipl_alignment_check(void)
 	return -1;
 }
 
+/**
+ * @brief Initialise the clock chip and then check the status of all clocks
+ *
+ * For all clocks available, do a soft reset, and then check the status register
+ * to check whether it encounter a calibration failure or not.
+ *
+ * @param[out] clock_count    number of clocks available in the system
+ *
+ * @return 0 on success, 1 on failure
+ */
+static int initialize_and_check_clock_chip(uint8_t& clock_count)
+{
+	struct pdbg_target *clock_target;
+	enum pdbg_target_status status;
+	uint8_t data;
+	int rc = 0;
+	int i2c_rc = 0;
+	std::vector<std::pair<std::string, std::string>> ffdcs;
+	uint8_t clk_pos = 0;
+
+	// initialize output variable.
+	clock_count = 0;
+
+	pdbg_for_each_class_target("oscrefclk", clock_target) {
+		if (!ipl_is_functional(clock_target))
+			continue;
+
+		// clock count has to be incremented even if it failed to initiallize
+		clock_count++;
+		ffdcs.clear();
+
+		if(!pdbg_target_get_attribute(
+			clock_target, "ATTR_POSITION", 2, 1, &clk_pos)) {
+
+			ipl_log(IPL_ERROR, "Attribute ATTR_POSITION read failed"
+				" for clock '%s' \n", pdbg_target_path(clock_target));
+			ipl_error_callback(IPL_ERR_ATTR_ACCESS);
+			rc++;
+			continue;
+		}
+
+		status = pdbg_target_probe(clock_target);
```

Comment:
Firts add hwas based check if it is functional ionly continue probe function.

Reply:
i am doing this on line 454

Reply:
looks like some mess , when i looked earlier.

Reply:
Ok.